dmprd.py

```diff
@@ -60,10 +60,13 @@ def __init__(self, multicast_address: str, interface_address: str,
         if addrinfo[0] == socket.AF_INET:
             # IPv4 specific socket configuration
 
-            ip_mreqn = get_ip_mreqn_struct(multicast_address, interface_address,
-                                           interface_name)
-            self.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_IF,
-                            ip_mreqn)
+            # bind the socket to the right interface
+            # no idea why its 25, just found it somewhere in the internet and it works
+            self.setsockopt(socket.SOL_SOCKET, 25, interface_name.encode('utf-8'))
```

Comment:
quite risky. DId you try the code on a real world box with two interfaces? This should be tested/validated before merging. It is sufficient when you see on on wireshark that on both interfaces dmpr packets are broadcasted.

Reply:
until now, it was validated via mininet-emulation (host with multiple interfaces)
wireshark started on mininet host shows exactly this behaviour.
Due to the implementation of mininet, the result should be the same when executed natively on a physical machine